Export lora weights to sep file

lucylq opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #14756
* #14767

Differential Revision: [D83777195](https://our.internmc.facebook.com/intern/diff/D83777195/)

lucylq committed:
Export lora weights to sep file



Differential Revision: [D83777195](https://our.internmc.facebook.com/intern/diff/D83777195/)

[ghstack-poisoned]

lucylq committed:
Update on "Export lora weights to sep file"


Differential Revision: [D83777195](https://our.internmc.facebook.com/intern/diff/D83777195/)

[ghstack-poisoned]

lucylq committed:
Update on "[wip] Export lora weights to sep file"


Differential Revision: [D83777195](https://our.internmc.facebook.com/intern/diff/D83777195/)

[ghstack-poisoned]

lucylq committed:
Update on "[wip] Export lora weights to sep file"


Differential Revision: [D83777195](https://our.internmc.facebook.com/intern/diff/D83777195/)

[ghstack-poisoned]

lucylq committed:
Update on "[wip] Export lora weights to sep file"


Differential Revision: [D83777195](https://our.internmc.facebook.com/intern/diff/D83777195/)

[ghstack-poisoned]

lucylq committed:
Update on "[wip] Export lora weights to sep file"


Differential Revision: [D83777195](https://our.internmc.facebook.com/intern/diff/D83777195/)

[ghstack-poisoned]

shoumikhin commented on examples/models/llama/main.cpp:
```diff
@@ -8,6 +8,8 @@
 
 #include <gflags/gflags.h>
+#include <sstream>
+#include <vector>
```
nit: maybe some of these are already included by runer.h, etc

lucylq committed:
Update on "[wip] Export lora weights to sep file"


Differential Revision: [D83777195](https://our.internmc.facebook.com/intern/diff/D83777195/)

[ghstack-poisoned]

lucylq committed:
Update on "Export lora weights to sep file"


Differential Revision: [D83777195](https://our.internmc.facebook.com/intern/diff/D83777195/)

[ghstack-poisoned]

lucylq committed:
Update on "Export lora weights to sep file"


Differential Revision: [D83777195](https://our.internmc.facebook.com/intern/diff/D83777195/)

[ghstack-poisoned]

lucylq committed:
Update on "Export lora weights to sep file"


Differential Revision: [D83777195](https://our.internmc.facebook.com/intern/diff/D83777195/)

[ghstack-poisoned]

lucylq committed:
Update on "Export lora weights to sep file"


Differential Revision: [D83777195](https://our.internmc.facebook.com/intern/diff/D83777195/)

[ghstack-poisoned]

lucylq committed:
Update on "Export lora weights to sep file"


Differential Revision: [D83777195](https://our.internmc.facebook.com/intern/diff/D83777195/)

[ghstack-poisoned]

lucylq committed:
Update on "Export lora weights to sep file"


Differential Revision: [D83777195](https://our.internmc.facebook.com/intern/diff/D83777195/)

[ghstack-poisoned]

lucylq committed:
Update on "Export lora weights to sep file"


Differential Revision: [D83777195](https://our.internmc.facebook.com/intern/diff/D83777195/)

[ghstack-poisoned]

lucylq committed:
Update on "Export lora weights to sep file"


Differential Revision: [D83777195](https://our.internmc.facebook.com/intern/diff/D83777195/)

[ghstack-poisoned]
